### PR TITLE
Listener scope

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes https://github.com/stackabletech/listener-operator/issues/95

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: deploy/helm/secret-operator/templates/daemonset.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ spec:
           env:
             - name: CSI_ENDPOINT
               value: /csi/csi.sock
+            - name: LISTOP
+              value: /csi-listop/csi.sock
```
[the remark this answers is not in the document]
Comment (nightkr) [reply deploy/helm/secret-operator/templates/daemonset.yaml:40]: Good catch, legacy leftover. We used to query listop directly before `PodListeners` was added.

File: rust/operator-binary/src/backend/pod_info.rs [hunk 2]
```diff
@@ -71,15 +146,225 @@ impl PodInfo {
...
+                    SecretScope::ListenerVolume { name } => Some(name),
+                    _ => None,
+                }
+            })
+            .map(|listener_volume| {
+                Ok((
+                    listener_volume,
+                    volume_pvc_names
+                        .get(listener_volume)
+                        .context(GetListenerVolumeSnafu { listener_volume })?,
```
[the remark this answers is not in the document]
Comment (nightkr) [reply rust/operator-binary/src/backend/pod_info.rs:225]: We don't necessarily know the `storageClassName`. We preinstall one named `listeners.stackable.tech` (and assume it exists), but technically any SC with `StorageClass.provisioner: listeners.stackable.tech` would work.

[the remark this answers is not in the document]
Comment (siegfriedweber) [reply rust/operator-binary/src/backend/pod_info.rs:225]: Okay, then let's leave it as it is.

File: deploy/helm/secret-operator/templates/daemonset.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nightkr) [reply deploy/helm/secret-operator/templates/daemonset.yaml]: ebcda1af8811b53cb26b5c8ccd762b8a5ff87854

File: Cargo.toml [hunk 3]
```diff
@@ -51,5 +51,6 @@ yasna = "0.5"
 # Workaround for https://github.com/hyperium/tonic/issues/243
 h2 = { git = "https://github.com/stackabletech/h2.git", branch = "feature/grpc-uds" }
 
-# [patch."https://github.com/stackabletech/operator-rs.git"]
-# stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+[patch."https://github.com/stackabletech/operator-rs.git"]
+# stackable-operator = { path = "../operator-rs" }
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "feature/podlisteners" }
```
[the remark this answers is not in the document]
Comment (nightkr) [reply Cargo.toml]: 9ad9cfc083caab3a02d9b2398cf7b4729673b808

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,16 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Added listener scope for provisioned secrets ([#310]).
+
+### Changed
+
+- `operator-rs` `0.44.0` -> `0.48.0` ([#310]).
```
[the remark this answers is not in the document]
Comment (nightkr) [reply CHANGELOG.md]: 9ad9cfc083caab3a02d9b2398cf7b4729673b808

File: docs/modules/secret-operator/pages/scope.adoc
[the remark this answers is not in the document]
Comment (nightkr) [reply docs/modules/secret-operator/pages/scope.adoc:1]: e7f11e033fa4ad919e4dcb842b559ace544ddf87

**Comment by siegfriedweber** [comment]
Waiting for https://github.com/stackabletech/listener-operator/pull/100#discussion_r1323254091 to be resolved. But otherwise LGTM.

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**